### PR TITLE
Гусев Никита. Вариант 17. Технология STL. Поразрядная сортировка для целых чисел с простым слиянием 

### PR DESCRIPTION
**Pull request by Cufeed**
Массив разделяется на отрицательные (хранятся как положительные) и положительные числа
Каждый поток обрабатывает свой чанк исходного массива (SplitAndSort: разделение массива на чанки для потоков)
Отрицательные сортируются как положительные, разворачиваются, восстанавливается знак
Положительные сортируются напрямую
Параллельная сортировка отрицательных и положительных частей выполняется в отдельных потоках
Для каждого разряда вызывается CountingSort, подсчет цифр в текущем разряде распределяется между потоками

### CODE REVIEW

File: tasks/stl/gusev_n_sorting_int_simple_merging/func_tests/main.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+TEST(gusev_n_sorting_int_simple_merging_stl, test_radix_sort_duplicates) {
+  std::vector<int> in = {5, 3, 5, -2, 3, -2, -2, 5, 0, 0, 7, 7, -7};
+  std::vector<int> out(in.size());
+  RunTest(in, out);
+  std::vector<int> expected = in;
+  std::ranges::sort(expected);
+  EXPECT_EQ(expected, out);
+}
+
+TEST(gusev_n_sorting_int_simple_merging_stl, test_radix_sort_reverse_order) {
```
Comment (petr1core) [inline tasks/stl/gusev_n_sorting_int_simple_merging/func_tests/main.cpp:72]: Add tests for vector with same values, single zero value (will not be superfluous, but you already have one with forty-two), all negative values, random generated values

**Review by petr1core — CHANGES_REQUESTED** [review]

File: tasks/stl/gusev_n_sorting_int_simple_merging/func_tests/main.cpp (hunk 1 above)
Comment (Cufeed) [reply tasks/stl/gusev_n_sorting_int_simple_merging/func_tests/main.cpp:111]: Done

File: tasks/stl/gusev_n_sorting_int_simple_merging/data/test.txt
[the remark this answers is not in the document]
Comment (Cufeed) [reply tasks/stl/gusev_n_sorting_int_simple_merging/data/test.txt:1]: Done

**Review by ermolaevv — APPROVED** [review]

**Review by petr1core — APPROVED** [review]

**Comment by Cufeed** [comment]
Добавил изменения [ревью](https://github.com/learning-process/ppc-2025-threads/pull/347/commits/978514e800dcd85ece40e685a198ef3a299f5abf) из TBB PR, которое включает в себя изменение STL PR (Все генераторы под один вид)

**Review by aobolensk — APPROVED** [review]

**Review by allnes — APPROVED** [review]
